psm_lib.py

```
@@ -16,53 +41,335 @@ def calculate_propensity_score(df, treatment_col, covariate_cols):
+        fig.update_layout(
+            title="Love Plot: Standardized Mean Difference (SMD)",
+            xaxis_title="Absolute Standardized Mean Difference",
+            yaxis_title="Variables",
+            legend_title="Status",
+            margin={'l': 0, 'r': 0, 't': 40, 'b': 0},
+            height=max(400, len(df_plot) * 30), # Auto-height based on number of vars
+            template="plotly_white"
+        )
+        return fig
+
+    except Exception as e:
```

Ruff (0.14.10)

Local variable `e` is assigned to but never used

Remove assignment to unused variable `e`

(F841)